Mathlib/Topology/CWComplex.lean

```diff
@@ -0,0 +1,164 @@
+/-
+Copyright (c) 2024 Elliot Dean Young and Jiazhen Xia. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiazhen Xia, Elliot Dean Young
+-/
+
+import Mathlib.Topology.Category.TopCat.Limits.Basic
+import Mathlib.Analysis.InnerProductSpace.PiL2
+
+/-!
+# CW-complexes
+
+In this file we define a relative CW-complex as the colimit of an expanding sequence of subspaces
+`sk i` (called the `i`-skeleta) for `i ≥ -1`, where `sk (-1)` is an arbitrary topological space
+and each `sk (n+1)` is obtained from `sk n` by attaching (n+1)-disks.
+
+## References
+The definition of CW complexes follows David Wärn's suggestion at
+https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Do.20we.20have.20CW.20complexes.3F/near/231769080
+-/
+
+open CategoryTheory
+
+namespace CWComplex
+
+/-- The n-dimensional sphere is the set of points in ℝ^(n+1) whose norm equals 1, endowed with the
+subspace topology. -/
+noncomputable def sphere (n : ℤ) : TopCat :=
+  TopCat.of <| Metric.sphere (0 : EuclideanSpace ℝ <| Fin <| Int.toNat <| n + 1) 1
+
+/-- The n-dimensional closed disk is the set of points in ℝ^n whose norm is at most 1, endowed with
+the subspace topology. -/
+noncomputable def disk (n : ℤ) : TopCat :=
+  TopCat.of <| Metric.closedBall (0 : EuclideanSpace ℝ <| Fin <| Int.toNat n) 1
+
+/-- `𝕊 n` denotes the n-dimensional sphere. -/
+notation "𝕊 "n => sphere n
```

Comment:
make these notations local, or better: scoped and move them to the right place.

Reply:
Thanks. Now they are scoped notations.